torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -507,6 +507,7 @@ def speculate_subgraph(
     should_flatten_outputs=False,
     under_activation_checkpoint=False,
     supports_input_mutation=True,
+    supports_aliasing=True,
```

Comment:
Can the default for this be False? Also ideally we would go through all HOPs that use the _has_potential_branch_input_alias check and update them to use this (could you file an issue for that?)

Reply:
I am not sure if the default should be `False`. For example, for activation checkpointing, it might be ok with aliasing? I will open an issue to track for sure.

Reply:
activation checkpointing is OK with it. The way I'd pitch this is: "supports_aliasing=False" is a safer default, because otherwise, someone writing a new HOP may run into silent incorrectness. For example, if someone is writing invoke_subgraph2, if they set supports_aliasing=True but do not handle it separately, they will get silen tincorrectness.